pkg_resources/__init__.py

```diff
@@ -70,15 +70,14 @@
     drop_comment,
     join_continuation,
 )
-
-from pkg_resources.extern import platformdirs
-from pkg_resources.extern import packaging
-
-__import__('pkg_resources.extern.packaging.version')
-__import__('pkg_resources.extern.packaging.specifiers')
-__import__('pkg_resources.extern.packaging.requirements')
-__import__('pkg_resources.extern.packaging.markers')
-__import__('pkg_resources.extern.packaging.utils')
+from pkg_resources.extern.packaging.markers import Marker, InvalidMarker
+from pkg_resources.extern.packaging.requirements import (
+    InvalidRequirement as _packaging_InvalidRequirement,
+    Requirement as _packaging_Requirement,
+)
+from pkg_resources.extern.packaging.utils import canonicalize_name
+from pkg_resources.extern.packaging.version import Version, InvalidVersion
+from pkg_resources.extern.platformdirs import user_cache_dir
```

Comment:
I know that `pkg_resources` does have `__all__` set, but I think I would worry less of people trying to import these names directly from `pkg_resources` if they had the `_` char?

Maybe something like:

```diff
-__import__('pkg_resources.extern.packaging.version')
-__import__('pkg_resources.extern.packaging.specifiers')
-__import__('pkg_resources.extern.packaging.requirements')
-__import__('pkg_resources.extern.packaging.markers')
-__import__('pkg_resources.extern.packaging.utils')
+from .extern.packaging import version as _packaging_version
+from .extern.packaging import specifiers as _packaging_specifiers
+from .extern.packaging import requirements as _packaging_requirements
+from .extern.packaging import markers as _packaging_markers
+from .extern.packaging import utils as _packaging_utils
```

(The rest we probably can leave as it is, right?)

Reply:
I see no issue aliasing packaging imports to private names.